### PR TITLE
Update status sections to include additional statuses

### PR DESCRIPTION
**Pull request by whabanks**
# Summary | Résumé
This PR updates our API documentation to reflect the new `status_description` being added to the API as well as callbacks: https://github.com/cds-snc/notification-api/pull/1875. Additionally it adds a few additional statuses and descriptions that were missing.

### CODE REVIEW

File: src/en/status.md [hunk 1]
```diff
@@ -60,15 +61,16 @@ If the request is successful, the response body is `json` and the status code is
   "email_address": "sender@something.com",  # required string for emails
   "phone_number": "+447900900123",  # required string for text messages
```
Comment (jimleroyer) [inline src/en/status.md:62]: We want to get rid of these comments? that says what's required and for which notification types. Do we have other documentation that covers this?

Comment (whabanks) [reply src/en/status.md:62]: I have not removed the comments throughout. Github is highlighting them red but a slightly different shade of red compared to actual removals. Perhaps Github is smart enough to know that the code block is marked as JSON which doesn't support comments?

Comment (jimleroyer) [reply src/en/status.md:62]: oh alright! sorry about that 😅

**Review by jimleroyer — APPROVED** [review]

File: src/en/status.md (hunk 1 above)
Comment (whabanks) [reply src/en/status.md:62]: All good, it caught me off guard the first time I looked at it too!

**Pull request merged**